controllers/metal3.io/host_state_machine.go

```diff
@@ -417,6 +417,13 @@ func (hsm *hostStateMachine) handleAvailable(info *reconcileInfo) actionResult {
 	} else if dirty {
 		hsm.NextState = metal3v1alpha1.StatePreparing
 		return actionComplete{}
+	} else {
+		// Check if hostFirmwareSettings have been changed
+		dirty, _, _ := hsm.Reconciler.getHostFirmwareSettings(info)
```

Comment:
I don't think we should ignore `err` here.

Reply:
Fixed